### PR TITLE
Split GLabs Traffic Driver ("Carrot") into two different slots: inline and left.

### PR DESCRIPTION
**Pull request by JonNorman**
## What does this change?
This replaces the "carrot" ad slot with two variants:
1. glabs-inline: this is a redesigned version of the original carrot.
2. glabs-left: this is essentially a rich-link, but served into an ad slot via DFP.

The two designs are implemented [here](https://github.com/guardian/commercial-templates/tree/master/src/glabs-inline) and [here](https://github.com/guardian/commercial-templates/tree/master/src/glabs-left) for inline and left respectively.

This remains a 0% ab test.

**Screenshots**:
*Inline*
image

*Left*
image

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    element.textContent.split(' ').length;
+
+const rules = {
+    bodySelector: '.js-article__body',
+    slotSelector: ' > p',
+    minAbove: 0,
+    minBelow: 0,
+    clearContentMeta: 0,
+    selectors: null,
+    filter: (slot: Object, index: number) => {
```
Comment (gustavpursche) [inline static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js]: The return type is missing.

File: static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        runningWordCount += wordCount(slot.element);
+        return index >= 2 && runningWordCount >= 100;
+    },
+};
+
+const insertInlineSlot = (paras: Element[]): Promise<any> => {
```
Comment (gustavpursche) [inline static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js]: Shouldn't the return type be `Promise<void>` (`addSlot()` doesn't return anything)?

File: static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            waitForLinks: false,
+            waitForInteractives: false,
+        });
+    } else if (commercialFeatures.glabsTrafficDriverLeftSlot) {
+        return spaceFiller.fillSpace(rules, insertLeftSlot, {
+            waitForImages: false,
+            waitForLinks: false,
+            waitForInteractives: false,
+        });
+    }
```
Comment (gustavpursche) [inline static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js]: Would it be better to write it:

```js
let slot;

if (commercialFeatures.glabsTrafficDriverInlineSlot) {
    slot = insertInlineSlot;
} else if (commercialFeatures.glabsTrafficDriverLeftSlot) {
   slot = insertLeftSlot;
}

if (slot) {
    return spaceFiller.fillSpace(rules, slot, {
        waitForImages: false,
        waitForLinks: false,
        waitForInteractives: false,
    });
}

?

File: static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+const insertLeftSlot = (paras: Element[]): Promise<any> => {
+    const slot = createSlot('glabs-left');
+    return fastdom
+        .write(() => paras[0].insertAdjacentElement('afterend', slot))
+        .then(() => addSlot(slot, true));
+};
+
+const glabsTrafficDriverSlotInit = () => {
```
Comment (gustavpursche) [inline static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js]: The return type is missing.

File: static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js (hunk 3 above)
Comment (JonNorman) [inline static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js]: Yep, much better.

**Comment by JonNorman** [comment]
[Push the new Glabs Traffic Drivers live.](https://trello.com/c/9QhGK7tT/202-push-the-new-glabs-traffic-drivers-live)

File: static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+const glabsTrafficDriverSlotInit = (): Promise<void> => {
+    let insertSlot;
+
+    if (commercialFeatures.glabsTrafficDriverInlineSlot) {
+        insertSlot = insertInlineSlot;
+    } else if (commercialFeatures.glabsTrafficDriverLeftSlot) {
+        insertSlot = insertLeftSlot;
+    }
+
+    if (!insertSlot) {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/glabs-traffic-driver-slot.js]: `else {}`

**Review by rich-nguyen — APPROVED** [review]

**Pull request merged**